src/generation.py

```diff
@@ -128,10 +141,16 @@ def generate(
                 x = torch.cat([x, new_idx[:, None]], dim=-1)
 
         if verbose:
-            y = self.tokenizer.detokenize_batch(
-                generation[:, : i + 1],
-                skip_special_tokens=skip_special_tokens,
-            )
+            if isinstance(self.tokenizer, CharLevelTokenizer):
```

Comment:
Maybe could slightly reformat this to have different args depending on the Tokenizer class

Reply:
Reformatted